### PR TITLE
Fix build and add gNOI versions.

### PR DESCRIPTION
**Pull request by robshakir**
Clean up gNOI repository for clarity and standard Go build.
 * Remove all BUILD.bazel files from the repository, since there are no users currently.
 * Move types and common protos to separate directories to work with the standard Go build system.
 * Add a gnoi_version proto FileOption such that we can mark the version of all files, and assign 0.1.0 as the first version at this point.

### CODE REVIEW

**Comment by robshakir** [comment]
@ejbrever @samribeiro @aashaikh -- PTAL, this PR makes some cleanups to the gNOI repository. One backwards incompatible change is moving types.proto -- please review the impact of this in your projects.

File: test/simple_test.go [hunk 1]
```diff
@@ -28,16 +28,16 @@ func TestGNOI(t *testing.T) {
...
-		desc: "gpb.Path",
-		in: &gpb.Path{
+		desc: "tpb.Path",
+		in: &tpb.Path{
 			Origin: "oc",
-			Elem:   []*gpb.PathElem{{Name: "interfaces", Key: map[string]string{"name": "Ethernet1/1/0"}}},
+			Elem:   []*tpb.PathElem{{Name: "interfaces", Key: map[string]string{"name": "Ethernet1/1/0"}}},
 		},
 		want: "origin: \"oc\"\nelem: <\n  name: \"interfaces\"\n  key: <\n    key: \"name\"\n    value: \"Ethernet1/1/0\"\n  >\n>\n",
 	}, {
```
Comment (levhuan) [inline test/simple_test.go]: Would you consider adding the test for system.SetPackage(RemoteDownload with Credentials)? This test requires simple_test.go to import both "gnoi/common" and "gnoi/types".

The code diff is listed below:

```
diff --git a/test/simple_test.go b/test/simple_test.go
index ecc708c..1b8405c 100644
--- a/test/simple_test.go
+++ b/test/simple_test.go
@@ -18,8 +18,10 @@ import (
 	"testing"
 
 	"github.com/golang/protobuf/proto"
-	gpb "github.com/openconfig/gnoi"
 	bgppb "github.com/openconfig/gnoi/bgp"
+	"github.com/openconfig/gnoi/common"
+	gSystem "github.com/openconfig/gnoi/system"
+	gpb "github.com/openconfig/gnoi/types"
 )
 
 func TestGNOI(t *testing.T) {
@@ -34,6 +36,19 @@ func TestGNOI(t *testing.T) {
 			Elem:   []*gpb.PathElem{{Name: "interfaces", Key: map[string]string{"name": "Ethernet1/1/0"}}},
 		},
 		want: "origin: \"oc\"\nelem: <\n  name: \"interfaces\"\n  key: <\n    key: \"name\"\n    value: \"Ethernet1/1/0\"\n  >\n>\n",
+	}, {
+		desc: "system.SetPackage",
+		in: &gSystem.Package{
+			Filename: "filename",
+			RemoteDownload: &gnoi.RemoteDownload{
+				Path:     "foo",
+				Protocol: gnoi.RemoteDownload_SCP,
+				Credentials: &gpb.Credentials{
+					Username: "bar",
+				},
+			},
+		},
+		want: "filename: \"filename\"\nremote_download: <\n  path: \"foo\"\n  protocol: SCP\n  credentials: <\n    username: \"bar\"\n  >\n>\n",
 	}, {
 		desc: "gpb.HashType",
 		in: &gpb.HashType{
```

File: common/common.proto [hunk 2]
```diff
@@ -18,7 +18,7 @@ syntax = "proto3";
 
 package gnoi;
```
[the remark this answers is not in the document]
Comment (robshakir) [reply common/common.proto]: Done -- renamed to `types` and `common` such that we don't have a `gnoi` package at all.

File: test/simple_test.go (hunk 1 above)
Comment (robshakir) [reply test/simple_test.go]: Added this test case, thanks!

**Comment by robshakir** [comment]
Thanks for the reviews here folks. Comments addressed, so I'll merge after the newly-setup travis run completes.

**Pull request merged**